hera_qm/utils.py

```diff
@@ -722,10 +724,13 @@ def apply_yaml_flags(uv, a_priori_flag_yaml, lat_lon_alt_degrees=None, telescope
             antnum = ant[0]
         if issubclass(uv.__class__, UVData):
             blt_selection = np.logical_or(uv.ant_1_array == antnum, uv.ant_2_array == antnum)
-            uv.flag_array[blt_selection, :, :, pol_selection] = True
+            if np.count_nonzero(blt_selection) > 0:
+                for bltind in np.where(blt_selection)[0]:
+                    uv.flag_array[bltind, :, :, pol_selection] = True
         elif issubclass(uv.__class__, UVCal):
             ant_selection = uv.ant_array == antnum
-            uv.flag_array[ant_selection, :, :, :, pol_selection] = True
-
+            if np.count_nonzero(ant_selection) > 0:
+                for antind in np.where(ant_selection)[0]:
```

Comment:
Is there a reason you can't do `uv.flag_array[ant_selection, :, :, :, pol_selection] = True`?

Reply:
I changed to this approach because `numpy` was throwing errors when the slice lengths don't match. This may be a version specific problem. With this change it is no longer an issue.